presto-main/src/main/java/io/prestosql/security/CatalogAccessControlRule.java

```diff
@@ -24,26 +24,38 @@
 public class CatalogAccessControlRule
 {
     private final boolean allow;
+    private final boolean readOnly;
     private final Optional<Pattern> userRegex;
     private final Optional<Pattern> catalogRegex;
 
     @JsonCreator
     public CatalogAccessControlRule(
             @JsonProperty("allow") boolean allow,
+            @JsonProperty("readOnly") boolean readOnly,
```

Comment:
Can we think of any other access modes in the future? If yes, this would be an enum.

Then in JSON this could look like:
```
   "access": "READ_ONLY",
```

Reply:
cc @electrum

Reply:
I didn't come up with any future extensions so in JSON this may be represented as `"readOnly": true` (as implemented), but maybe @electrum has some other thoughts.